OsmAnd-java/src/main/java/net/osmand/router/network/NetworkRouteSelector.java

```diff
@@ -549,6 +550,28 @@ private void loadData(NetworkRouteSegment segment, RouteKey rkey, List<NetworkRo
 		}
 	}
 
+	private void addSegmentsBiggerThanOneTile(Map<RouteKey, List<NetworkRouteSegment>> tiles,
+	                                          long tileId, TLongArrayList queue) {
+		for (Map.Entry<RouteKey, List<NetworkRouteSegment>> entry : tiles.entrySet()) {
+			for (NetworkRouteSegment segment : entry.getValue()) {
+				RouteDataObject rdo = segment.robj;
+				if (rdo == null) {
+					continue;
+				}
+				long startPointTileId = NetworkRouteContext.getTileId(rdo.getPoint31XTile(segment.start), rdo.getPoint31YTile(segment.start));
+				long endPointTileId = NetworkRouteContext.getTileId(rdo.getPoint31XTile(segment.end), rdo.getPoint31YTile(segment.end));
+				if (queue != null) {
+					if (tileId != startPointTileId) {
```

Comment:
Missing ? 
if (!visitedTiles.add(tileID)) {
				continue;
			}

Reply:
This check in the begin of next while cycle (line 531)

Reply:
Agree with Dima